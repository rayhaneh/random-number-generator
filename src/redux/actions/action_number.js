import { get } from 'axios';

export const FETCH_NUMBER = 'FETCH_NUMBER'
export const fetchNumber = (range) => {
  const API_BASE_URL='https://j160x3dmd0.execute-api.us-west-2.amazonaws.com/dev';
  const url = `${ API_BASE_URL }/numbers?min=${ range.min }&max=${ range.max }`
  return get(url)
    .then(response => {
      return {
        type: FETCH_NUMBER,
        payload: response
      }
    });
}
