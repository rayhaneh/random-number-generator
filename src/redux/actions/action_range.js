export const CHANGE_MAX_VALUE = 'CHANGE_MAX_VALUE';
export const changeMax = (maxValue) => {
  return {
    type: CHANGE_MAX_VALUE,
    payload: Number(maxValue),
  }
}

export const CHANGE_MIN_VALUE = 'CHANGE_MIN_VALUE';
export const changeMin = (minValue) => {
  return {
    type: CHANGE_MIN_VALUE,
    payload: Number(minValue),
  }
}