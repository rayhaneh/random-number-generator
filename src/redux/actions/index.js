export { fetchNumber, FETCH_NUMBER } from './action_number';

export {
  changeMax,
  CHANGE_MAX_VALUE,
  changeMin,
  CHANGE_MIN_VALUE,
} from './action_range';