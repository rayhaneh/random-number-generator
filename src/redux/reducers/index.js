import { combineReducers } from 'redux';
import { NumberReducer } from './reducer_number';
import { RangeReducer } from './reducer_range';

const rootReducer = combineReducers({
  number: NumberReducer,
  range: RangeReducer,
});

export default rootReducer;
