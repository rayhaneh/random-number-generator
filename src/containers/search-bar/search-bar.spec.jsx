import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import {
  SearchBar,
  mapStateToProps,
  mapDispatchToProps,
} from './search-bar';
import { CHANGE_MIN_VALUE } from '../../redux/actions';

let props;
let comp;
let mockMaxChange = jest.fn();
let mockMinChange = jest.fn();
let mockFetchNumber = jest.fn();

describe('SearchBar Container', () => {
  beforeEach(() => {
    props = {
      range: {
        max: 50,
        min: 0,
      },
      changeMin: mockMinChange,
      changeMax: mockMaxChange,
      fetchNumber: mockFetchNumber,
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders properly', () => {
    comp = shallow(<SearchBar {...props} />);
    expect(comp).toMatchSnapshot();
  });

  it('should call change max action when onMaxChange method is called', () => {
    const instance = new SearchBar(props);
    expect(mockMaxChange.mock.calls.length).toBe(0);
    instance.onMaxChange({target: {value: 2000}})
    expect(mockMaxChange.mock.calls.length).toBe(1);
  });

  it('should call change min action when onMinChange method is called', () => {
    const instance = new SearchBar(props);
    expect(mockMinChange.mock.calls.length).toBe(0);
    instance.onMinChange({target: {value: 2}})
    expect(mockMinChange.mock.calls.length).toBe(1);
  });

  it('should call fetch number action when onSubmit method is called', () => {
    const instance = new SearchBar(props);
    expect(mockFetchNumber.mock.calls.length).toBe(0);
    instance.onSubmit({ preventDefault: jest.fn() })
    expect(mockFetchNumber.mock.calls.length).toBe(1);
  });

  it('should return range from the state', () => {
    const state = {
      range: {
        max: 1000,
        min: 1,
      },
      number: 200,
    };

    expect(mapStateToProps(state)).toEqual({
      range: state.range,
    });
  });

  it('should return range from the state', () => {
    const mockDispatch = jest.fn();

    mapDispatchToProps(mockDispatch).changeMin(10);
    expect(mockDispatch.mock.calls.length).toBe(1);
    expect(mockDispatch.mock.calls[0][0]).toEqual({ type: CHANGE_MIN_VALUE, payload: 10});
  });
});
