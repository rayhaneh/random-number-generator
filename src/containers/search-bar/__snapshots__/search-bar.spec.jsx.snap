// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Container renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    changeMax={[MockFunction]}
    changeMin={[MockFunction]}
    fetchNumber={[MockFunction]}
    range={
      Object {
        "max": 50,
        "min": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "max": 50,
      "min": 0,
      "onMaxChange": [Function],
      "onMinChange": [Function],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "max": 50,
        "min": 0,
        "onMaxChange": [Function],
        "onMinChange": [Function],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
